File: list-chunk/list-chunk.py

import random

def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]


nums = [random.uniform(1, 10) for i in range(10)]


nums_groups = list(chunks(nums, 4))


print(nums_groups)

File: list-chunk/list-chunk.js
function* chunks(nums, n) {
  for (let i = 0; i < nums.length; i += n) {
    yield nums.slice(i, i + n);
  }
}

nums = [];
while (nums.length < 10) {
  var r = Math.floor(Math.random() * 100) + 1;
  if (nums.indexOf(r) === -1) nums.push(r);
}

console.log(nums);
// spread generator 
nums_groups = [...chunks(nums, 4)];

console.log(nums_groups);
